src/cli/serve/serve.js

```diff
@@ -44,8 +44,26 @@ const getBootstrapScript = (isDev) => {
   }
 };
 
-const setServerlessKibanaDevServiceAccountIfPossible = (set, opts) => {
-  if (!opts.dev || !opts.serverless || process.env.isDevCliChild === 'true') {
```

Comment:
@mistic in #165357 added `process.env.isDevCliChild === 'true'` but this is always returning `true` locally so it blocks the service token when running ES and KBN locally with `--ssl`